lib/rules/no-js-extension.js

```diff
@@ -19,12 +19,36 @@ var util = require("../util");
 module.exports = function (context) {
 
     var MESSAGE = "Don't use .js extension in dependency path.";
+    var PLUGIN_PREFIX_REGEX = /^(\w+)\!/i;
```

Comment:
I'm wondering if you could build the regex once dynamically based on pluginsToCheck? So for a plugin list like ['foo', 'bar'] you might end up with /^(foo|bar)\!/i. This might allow you to get rid of the loop in `shouldBeChecked`